lib/auth/auth.go

```diff
@@ -1875,13 +1875,18 @@ func (a *Server) SetAccessRequestState(ctx context.Context, params services.Acce
 	if err := a.DynamicAccessExt.SetAccessRequestState(ctx, params); err != nil {
 		return trace.Wrap(err)
 	}
+	req, err := a.getAccessRequestByID(ctx, params.RequestID)
+	if err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
Setting the access request and then loading it separately may potentially cause inconsistent behavior.  The backend loads the access request in order to update it, so we actually have access to the exact copy of the modified request, it just isn't being returned right now.  I would suggest splitting up the `SetAccessRequestState` method into external and internal variants, so that the backend can return the exact updated request to `auth.Server`.

To see how this kind of division is done, take a look at the review submission methods.  The `DynamicAccess` interface (i.e. the client interface) implements the "public" version of the review submission method, while the `DynamicAccessExt` interface (i.e. the backend interface) implements a different variant that the auth server uses to pass in extra info.

In this case, we want to do the opposite: have the backend pass extra info back up to the auth server layer.